Xamarin/Droid/Screens/LoginDialog.cs

```diff
@@ -77,14 +91,33 @@ public override void OnStart()
             {
                 DrawXSettingsManager.Write(() =>
                 {
-                    DrawXSettingsManager.Settings.ServerIP = serverAddressEntry.Text;
-                    DrawXSettingsManager.Settings.Username = usernameEntry.Text;
+                    DrawXSettingsManager.Settings.ServerIP = _serverAddressEntry.Text;
+                    DrawXSettingsManager.Settings.Username = _usernameEntry.Text;
                 });
 
-                var credentials = Credentials.UsernamePassword(usernameEntry.Text, passwordEntry.Text, false);
+                Credentials credentials = null;
+                if (v == _loginButton)
+                {
+                    credentials = Credentials.UsernamePassword(_usernameEntry.Text, _passwordEntry.Text, false);
+                }
+                else if (v == _adLoginButton)
+                {
+                    // TODO: verify that server url has been input
+
+                    var authContext = new AuthenticationContext(ADCredentials.CommonAuthority);
+                    var response = await authContext.AcquireTokenAsync("https://graph.windows.net", ADCredentials.ClientId, ADCredentials.RedirectUri, new PlatformParameters(_parentActivity));
+
+                    // TODO: uncomment when implemented
+                    // var credentials = Credentials.ActiveDirectory(response.AccessToken);
+                    credentials = Credentials.Debug();
+                }
 
                 await PerformLoginAsync(credentials);
             }
+            catch (Exception ex)
+            {
+                // TODO: handle
```

Comment:
Is this meant to be a catch-all Exception handler or primarily for failure to get AD credentials?